pf/internal/schemashim/custom_object_type_test.go

```diff
@@ -0,0 +1,212 @@
+// Copyright 2016-2023, Pulumi Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package schemashim
+
+import (
+	"context"
+	"fmt"
+	"reflect"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-framework/attr"
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-framework/types/basetypes"
+	"github.com/hashicorp/terraform-plugin-go/tftypes"
+	"github.com/stretchr/testify/assert"
+
+	"github.com/pulumi/pulumi-terraform-bridge/v3/pkg/tfshim"
+)
+
+func TestCustomObject(t *testing.T) {
+	ctx := context.Background()
+
+	obj := newObjectPseudoResource(NewObjectTypeOf[SomeType](ctx), nil, nil)
+
+	s := obj.Schema().Get("s")
+	assert.Equal(t, shim.TypeString, s.Type())
+}
+
+type SomeType struct {
+	S types.String `tfsdk:"s"`
+}
+
+// --- custom object machinery ---
+//
+// copied from terraform-provider-aws/internal/framework/types/objectof.go
```

Comment:
Is it reasonable to use the TF AWS implementation of custom object as a general test support machinery here? unless I'm misunderstanding something, this pattern is specific to AWS custom objects, isn't it?

I wonder if it would be possible (and not awfully impractical) to pull this in as a dependency instead, rather than copying?

Reply:
I think it is.